### ISSUE TITLE
Error merging into Optional[List[int]] = None

### ISSUE THREAD

**Issue by omry** [omry/omegaconf#428]
repro:
```python
from dataclasses import dataclass
from typing import Optional, List
from omegaconf import OmegaConf


@dataclass
class Config:
    my_field: Optional[List[int]] = None


OmegaConf.merge(Config, Config)
```

This is a bug in 2.0 and also in master (and should be fixed in both).

**Comment by omry** [comment]
Fixed in 19da4a404d15218ee592d7048a2008f7d7062fdf for 2.0.
2.0.4 coming.

**Comment by omry** [comment]
#430 fixes in master.
Released 2.1.0-dev10 with this fix.
